### PR TITLE
Add weekly build and release process

### PR DESCRIPTION
**Pull request by Zhangbaowen-Hashdata**
We'll schedule weekly builds and releases to start every Monday morning. 
During this process, we'll compile the CBDB RPM package from its source 
and tag the latest commit to generate a release note automatically.

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: deploy/build/README.CentOS.bash [hunk 1]
```diff
@@ -31,7 +31,7 @@ sudo yum install -y \
     postgresql-devel
 
 sudo pip3 install conan
-sudo pip3 install -r python-dependencies.txt
+sudo pip3 install -r $GITHUB_WORKSPACE/python-dependencies.txt
```
Comment (tuhaihe) [inline deploy/build/README.CentOS.bash:34]: Don't change this path.

**Review by tuhaihe — CHANGES_REQUESTED** [review]

File: .github/workflows/cbdb_release_weekly.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+name: Build and Release RPM Weeklyly
```
Comment (tuhaihe) [inline .github/workflows/cbdb_release_weekly.yml:1]: Weeklyly -> Weekly

File: deploy/build/README.CentOS.bash (hunk 1 above)
Comment (Zhangbaowen-Hashdata) [reply deploy/build/README.CentOS.bash:34]: README.CentOS needs to use python-dependencies.txt during execution. Before moving python-dependencies.txt, you can get it. But now, if you do not modify the location, the script cannot get the location of python-dependencies.txt. An error will be reported

File: .github/workflows/cbdb_release_weekly.yml (hunk 2 above)
Comment (Zhangbaowen-Hashdata) [reply .github/workflows/cbdb_release_weekly.yml:1]: Thanks for your correction, I will modify it

File: deploy/build/README.CentOS.bash (hunk 1 above)
Comment (tuhaihe) [reply deploy/build/README.CentOS.bash]: Ok, got it. Please update the file path to the latest path. Do not use variable $GITHUB_WORKSPACE, which is not defined in the bash script.

**Review by baotingfang — APPROVED** [review]

**Review by my-ship-it — APPROVED** [review]

**Review by tuhaihe — APPROVED** [review]

**Pull request merged**